activestorage/app/controllers/active_storage/representations/proxy_controller.rb

```diff
@@ -12,7 +12,7 @@ class ActiveStorage::Representations::ProxyController < ActiveStorage::Represent
 
   def show
     http_cache_forever public: true do
-      send_blob_stream @representation, disposition: params[:disposition]
+      send_blob_stream @representation, disposition: params[:disposition], filename: params[:download]
```

Comment:
I mentioned in the PR description why the parameter is named `download` instead of `filename`, but I'll re-share it here for more context:

---

The filename is derived from the `download` query parameter instead of a `filename` query parameter since the [Active Storage routes](https://github.com/rails/rails/blob/4df235f7c0149ac7be86580d41a74a7785cb9bb9/activestorage/config/routes.rb#L5-L11) utilize the `*filename` dynamic segment. At the controller level, the value of the `params[:filename]` path pattern collides with any `?filename=` query parameter. 

The `download` query parameter was chosen due to its correlation with the `<a>` element's `[download]` attribute:

```ruby
# before - might not download file to `avatar.png` depending on browser
link_to "Download", rails_blob_path(user.avatar, disposition: "attachment"), download: "avatar.png")

# after
link_to "Download", rails_blob_path(user.avatar, disposition: "attachment", download: "avatar.png")
```

As an alternative, the `disposition` query parameter could be re-purposed to support both String and Hash values:

```ruby
# before - might not download file to `avatar.png` depending on browser
link_to "Download", rails_blob_path(user.avatar, disposition: "attachment"), download: "avatar.png"

# after
link_to "Download", rails_blob_path(user.avatar, disposition: { attachment: "avatar.png" })
```

The trade-off with that developer experience is that passing `disposition: { inline: "avatar.png" }` would have no effect, and could surprise callers. While passing `:inline` as a key could output a warning, that's something that would require explicit intervention.

Reply:
Is naming this variable `filename` important enough justify re-naming the route dynamic segments in a separate PR ahead of this change?